Branch 1

YanethM opened the pull request.

juanhurtado10 commented on lib/hello_web/templates/hello/index.html.heex:
```diff
@@ -0,0 +1,3 @@
+<section class="phx-hero">
+  <h2>Hello World, from Phoenix!</h2>
+</section>
```
@YanethM missing an empty line at the end of the file. Tip: You can configure your code editor to automatically manages this for you

juanhurtado10 commented on lib/hello_web/router.ex:
```diff
@@ -18,6 +19,38 @@ defmodule HelloWeb.Router do
+    pipe_through :api
+
+    scope "/v1", V1, as: :v1 do
+      resources "/images",  ImageController
```
@YanethM what a `resources` do?

YanethM replied:
I associate it with the laravel framework that has a type of resource controller, although in phoenix they call it macro, which takes all HTTP requests and can handle only and except options.

juanhurtado10 commented on lib/hello_web/plugs/locale.ex:
```diff
@@ -0,0 +1,15 @@
+defmodule HelloWeb.Plugs.Locale do
```
@YanethM what is the benefit of using plugs?

YanethM replied:
As I understand it,  the plugs are used during the life cycle of HTTP actions. In this case, the use of plug is at the module level for the regional configuration of the language, but it can also be used in the router for some routes like match or dispatch.

juanhurtado10 commented on lib/hello_web/views/layout_view.ex:
```diff
@@ -1,7 +1,8 @@
-  # Phoenix LiveDashboard is available only in development by default,
-  # so we instruct Elixir to not warn if the dashboard route is missing.
-  @compile {:no_warn_undefined, {Routes, :live_dashboard_path, 2}}
+  def title() do
```
@YanethM by coding style, the functions with arity 0 should not include parentheses.

juanhurtado10 commented on lib/hello_web/templates/page/index.xml.eex:
```diff
@@ -0,0 +1,3 @@
+<user>
+    <name>Yaneth</name>
```
@YanethM Please make sure your code editor is properly indenting your code. 1 Tab == 2 spaces

juanhurtado10 commented on lib/hello_web/templates/page/index.text.eex:
```diff
@@ -0,0 +1 @@
+OMG, this is actually some text.
```
@YanethM missing blank line at the end of the file

juanhurtado10 commented on lib/hello_web/components/weather.ex:
```diff
@@ -0,0 +1,14 @@
+defmodule HelloWeb.Weather do
+  use Phoenix.Component
```
@YanethM what does a `Phoenix.Component` do?

YanethM replied:
A component is a function that can be called inside heex and can be in the same module or an external one, as in this case, I call the component in the show.html.heex view.

YanethM committed:
Remove parenthesis

Solved problems

Module 2: Basics

Create the project structure using Phoenix fm

Create the project structure using Phoenix fm

Solved problems

Solved problems

Create the project structure using Phoenix fm

Add blank line in html.heex files

Create the project structure using Phoenix fm
